learn/develop/sparse-data/index.qmd

```diff
@@ -0,0 +1,92 @@
+
+The sparsity mostly matters for the predictors. The outcomes, case weights, and predictions are done densely.
+
+Below is outlined how sparse data (matrix & tibble) works with the various tidymodels packages.
+
+## rsample
+
+The resampling functions from the rsample package work with sparse tibbles out of the box. However, they won't work with sparse matrices. Instead, use the `coerce_to_sparse_tibble()` function from the sparsevctrs package to turn the sparse matrix into a sparse tibble and proceed with that object.
+
+```r
+library(sparsevctrs)
+data_tbl <- coerce_to_sparse_tibble(data_mat)
+```
+
+## recipes
+
+The recipes package receives data at 3 different places, in `recipe()`, `prep()`, and `bake()`. These functions all handle sparse data in the same matter.
+
+Sparse tibbles should work as normal. Sparse matrices are accepted and then turned into sparse tibbles right away, and flow through the internals as sparse tibbles.
+
+The `composition` argument of `bake()` understands sparse tibbles. When `composition = "dgCMatrix"` the resulting sparse matrix will be created from the sparse tibble with minimal overhead.
+
+The different recipes steps don't know how to handle sparse vectors yet. This will mean that most likely it will result in the materialization of the selected variables. This is a known issue and is planned to be fixed.
+
+## parsnip
+
+The parsnip package receives data in 3 places, in `fit()`, `fit_xy()`, and `predict()`.
+
+Sparse data does not yet work with the formula interface for `fit()`. This is a known issue and is expected to be fixed long term.
+
+For the remaining interfaces for `fit()` and `fit_xy()`, both sparse tibbles and sparse matrices are supported. Sparse matrices are turned into sparse tibbles early on. When fitting a model in parsnip. It checks whether the engine supports sparse matrices using the `allow_sparse_x` specification. A warning is thrown if sparse data is passed to an engine that doesn't support it, informing the user of that fact and that the data will be converted to a dense representation.
+
+`predict()` works with sparse tibbles and sparse matrics, where sparse matrices are turned into sparse tibbles right away, and into the appropriate format before it is passed to the model engine.
+
+## workflows
+
+The workflows package receives data in 2 places, in `fit()` and `predict()`.
+
+Sparse data does not yet work with the formula interface for `fit()`. This is a known issue and is expected to be fixed long term.
+
+Both `fit()` and `predict()` work with sparse matrices and sparse tibbles. Turning sparse matrics into sparse tibbles at the earliest convenience. Most of the checking and functionality is delegated to recipes and parsnip.
```

Comment:
need to clarify with what preprocessors `fit()` works and make the second sentence a complete sentence